### ISSUE TITLE
Freeze dependencies in CI

### ISSUE THREAD

**Issue by llllllllll** [blaze/blaze#1600]

**Comment by kwmsmith** [comment]
@llllllllll agreed 100%.  I'm working on freezing the deps currently.  Working around the current build failures is slowing down development.

Because we have other projects that use Blaze with sometimes-overlapping dependencies, my preference is to use open-ended dependencies (i.e., `pymongo<=3.0` rather than `pymongo=3.0`) unless constrained otherwise.

**Comment by llllllllll** [comment]
For the CI I think they should truly be frozen to make sure that we can all reproduce a test setup. Even minor patches could affect behavior in an unexpected way. I think for installs of blaze|odo|datashape we should use soft constraints. This is what we do in zipline and has been very successful. Basically the requirements are all specified as `>=n`, the ci runs on one particular frozen env, and then iff we discover that something breaks on a newer version we add an upper bound.

**Comment by willianpaixao** [comment]
Is a "bump version bot" (like https://dependabot.com/) an option?
